gossipd/broadcast.c

```diff
@@ -88,6 +88,7 @@ const u8 *next_broadcast(struct broadcast_state *bstate,
 	return NULL;
 }
 
+#ifdef PEDANTIC
 static const struct pubkey *
```

Comment:
OK, the way CCAN does this is that the xxx_check function is always defined, and a wrapper xxx_debug() calls in if XXX_DEBUG is defined, otherwise it's a noop.  I'll implement that as a separate patch (we should really have it for all primitives we use, and then have a RPC call to do an internal consistency check).